src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,585 @@
+# Copyright (c) 2025: Andrew Rosemberg and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    fill_off_diagonal(H)
+
+Filling the off-diagonal elements of a sparse matrix to make it symmetric.
+"""
+function fill_off_diagonal(H::SparseMatrixCSC)
```

Comment:
we need underscores in all non-user facing API functions, e.g. `_fill_off_diagonal`

Reply:
underscores?